File: wtt_main_server/wtt_main_server_data_models/Database/Abstract/ADbHttpAction.cs
using wtt_main_server_data.Enums;
using HttpRequestMethod = wtt_main_server_data.Enums.HttpRequestMethod;

namespace wtt_main_server_data.Database.Abstract;

#pragma warning disable CS8618

public abstract class ADbHttpAction : ADbWebRequest
{
	/* Метод запроса к удаленному серверу.
	 * Основной набор: GET/POST/PUT/PATCH/DELETE
	 */
	public HttpRequestMethod Method { get; set; }
	
	public HttpTlsValidationMode TlsValidationMode { get; set; }
}

File: tmp/Actions/Abstract/ADbHttpAction.ts
import HttpRequestMethod from "src/models/Enums/HttpRequestMethod";
import ADbWebRequest from "./ADbWebRequest";
import HttpTlsValidationMode from "src/models/Enums/HttpTlsValidationMode";

export default abstract class ADbHttpAction extends ADbWebRequest
{
	method: HttpRequestMethod = HttpRequestMethod.Get;
	tlsValidationMode: HttpTlsValidationMode = HttpTlsValidationMode.Enabled;
}